aten/src/ATen/BatchingRegistrations.cpp

```diff
@@ -563,13 +563,11 @@ Tensor _new_zeros_with_same_feature_meta_batching_rule(
   return self_physical_view.getPhysicalToLogicalMap().apply(result);
 }
 
-bool _has_same_storage_numel_batching_rule(const Tensor& self, const Tensor& other) {
-  TORCH_CHECK(isBatchedTensor(self) && !isBatchedTensor(other),
-    "Only the 'batched grad' use case is supported in PyTorch core.");
-  // NB: This check doesn't really make sense because batched tensors
-  //     don't have storage. We should just forbid as_strided in
-  //     batched forward grad computations (TODO).
-  return true;
+int64_t _storage_numel_batching_rule(const Tensor& self) {
+  auto physical_view = at::MultiBatchVmapTransform::logicalToPhysical(self);
+  auto num_batch_dims = physical_view.numBatchDims();
+  const auto& physical_tensor = physical_view.tensor();
+  return physical_tensor.strides()[num_batch_dims - 1];
```

Comment:
This would only be valid if this is conrtiguous and the batch dims are first no?

Reply:
Ah yes, we'd need to add the check enforcing this.

`_new_zeros_with_same_feature_meta`  makes sure the the storage of the logical tensor as a whole is tiled contiguously, the logical tensor itself need not be contiguous